x-pack/plugins/aiops/public/components/explain_log_rate_spikes/field_filter_popover.tsx

```diff
@@ -0,0 +1,221 @@
+          onClick={onFieldSelectionButtonClick}
+          disabled={disabled}
+          size="s"
+          iconType="arrowDown"
+          iconSide="right"
+          iconSize="s"
+        >
+          <FormattedMessage
+            id="xpack.aiops.explainLogRateSpikesPage.fieldFilterButtonLabel"
+            defaultMessage="Filter fields"
+          />
+        </EuiButton>
+      }
+      isOpen={isFieldSelectionPopoverOpen}
+      closePopover={closePopover}
+    >
+      <EuiPopoverTitle>
+        <EuiText size="xs" color="subdued" style={{ maxWidth: '400px' }}>
+          <FormattedMessage
+            id="xpack.aiops.explainLogRateSpikesPage.fieldFilterHelpText"
+            defaultMessage="To remove non-relevant fields from groups, deselect them below and click the Apply button to rerun the grouping. The search bar can be used to filter the list of fields and is useful to select/deselect in bulk with the actions below the list."
+          />
+        </EuiText>
+        <EuiSpacer size="s" />
+        <EuiFieldText
+          compressed
+          placeholder={i18n.translate('xpack.aiops.analysis.fieldSelectorPlaceholder', {
+            defaultMessage: 'Search',
+          })}
+          aria-label={i18n.translate('xpack.aiops.analysis.fieldSelectorAriaLabel', {
+            defaultMessage: 'Filter fields',
+          })}
+          value={fieldSearchText}
+          onChange={(e: ChangeEvent<HTMLInputElement>) => setFieldSearchText(e.currentTarget.value)}
+          data-test-subj="aiopsFieldSelectorSearch"
+        />
+      </EuiPopoverTitle>
+      <div css={fieldSelectPopover} data-test-subj="aiopsFieldSelectorFieldNameList">
+        {filteredUniqueFieldNames.map((fieldName) => (
+          <div key={fieldName} css={{ padding: '4px' }}>
+            <EuiSwitch
+              data-test-subj={`aiopsFieldSelectorFieldNameListItem${
+                !skippedFields.includes(fieldName) ? ' checked' : ''
+              }`}
+              className="euiSwitch--mini"
+              compressed
+              label={fieldName}
+              onChange={(e) => setFieldsFilter([fieldName], e.target.checked)}
+              checked={!skippedFields.includes(fieldName)}
+            />
+          </div>
+        ))}
+      </div>
+      <EuiPopoverFooter>
+        <EuiFlexGroup
+          gutterSize="s"
+          responsive={false}
+          justifyContent="spaceBetween"
+          alignItems="center"
+        >
+          <>
+            <EuiFlexItem grow={false}>
+              <EuiButtonEmpty
+                size="xs"
+                flush="left"
+                onClick={() => setFieldsFilter(filteredUniqueFieldNames, true)}
+                data-test-subj="aiopsFieldSelectorEnableAllSelectedButton"
+              >
+                {fieldSearchText.length > 0 ? (
+                  <FormattedMessage
+                    id="xpack.aiops.explainLogRateSpikesPage.fieldSelector.enableAllSearchedFields"
```

Comment:
I think 'Select' and 'Deselect' would be better terms to use here - you are already using those terms in the help text and I think they are better than enable / disable.

Reply:
Fixed in [8b9f799](https://github.com/elastic/kibana/pull/153864/commits/8b9f7993e44226c2ce35d6ddb9fb986b3679ef6b).